unit-tests/src/test/java/org/eclipse/collections/impl/list/mutable/primitive/BoxedMutableBooleanListTest.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright (c) 2023 The Bank of New York Mellon.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v. 1.0 which accompany this distribution.
+ * The Eclipse Public License is available at http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ */
+
+package org.eclipse.collections.impl.list.mutable.primitive;
+
+import org.eclipse.collections.api.factory.Lists;
+import org.eclipse.collections.impl.test.Verify;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class BoxedMutableBooleanListTest
+{
+    private BoxedMutableBooleanList classUnderTest()
+    {
+        return new BoxedMutableBooleanList(new BooleanArrayList(true, true));
+    }
+
+    @Test
+    public void size()
+    {
+        BoxedMutableBooleanList list = this.classUnderTest();
+        Verify.assertSize(2, list);
+        list.add(false);
+        Verify.assertSize(3, list);
+    }
+
+    @Test
+    public void add()
+    {
+        BoxedMutableBooleanList list = this.classUnderTest();
+        list.add(Boolean.FALSE);
+        Assert.assertEquals(new BoxedMutableBooleanList(new BooleanArrayList(true, true, false)), list);
```

Comment:
I think this should be a JDK `ArrayList` or EC `FastList` of `Boolean` to make sure the equals contract works correctly.

Reply:
Fixed. Thanks.